### PR TITLE
ticdc: replica changelog to snowflake, add a part about ETL in snowflake

### PR DESCRIPTION
**Pull request by leoppro**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
LGTM

**Review by IANTHEREAL — APPROVED** [review]

File: ticdc/integrate-confluent-using-ticdc.md [hunk 1]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+      insert 
+          (i_data, i_id, i_im_id, i_name, i_price) 
+      values 
+          (v:i_data, v:i_id, v:i_im_id, v:i_name, v:i_price)
+;
+```
+
+在上面的语句中，我们使用了 Snowflake 的 MERGE INTO 语句，这个语句可以根据条件将流和表做匹配，然后根据不同的匹配结果，执行不同的操作，比如删除、更新或插入。在这个例子中，我们使用了三个 WHEN 子句，分别对应了三种情况：
+
+  * 当流和表匹配时，且流中的数据为空，则删除表中的记录
```
Comment (IANTHEREAL) [inline ticdc/integrate-confluent-using-ticdc.md:266]:
```suggestion
 * 当流和表匹配时，且流中的数据为空，则删除表中的记录
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 2]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+  * 当流和表匹配时，且流中的数据不为空，则更新表中的记录
```
Comment (IANTHEREAL) [inline ticdc/integrate-confluent-using-ticdc.md:267]:
```suggestion
- 当流和表匹配时，且流中的数据不为空，则更新表中的记录
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 3]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+  * 当流和表不匹配时，则插入流中的数据
```
Comment (IANTHEREAL) [inline ticdc/integrate-confluent-using-ticdc.md:268]:
```suggestion
- 当流和表不匹配时，则插入流中的数据
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 4]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+
+```
+create or replace table TEST_ITEM (
+    i_id INTEGER primary key,
+    i_data VARCHAR,
+    i_im_id INTEGER,
+    i_name VARCHAR,
+    i_price DECIMAL(36,2)
+);
+```
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:229]:
```suggestion
1. 根据 TiDB 中的表结构，在 Snowflake 中创建结构相同的表：

    ```
    create or replace table TEST_ITEM (
        i_id INTEGER primary key,
        i_data VARCHAR,
        i_im_id INTEGER,
        i_name VARCHAR,
        i_price DECIMAL(36,2)
    );
    ```
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 5]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+
+下面，为 TIDB_TEST_ITEM 创建 STREAM，将 append_only 设为 true，表示仅接收 INSERT 事件。这个 STREAM 可以实时的捕获 TIDB_TEST_ITEM 的 INSERT 事件，也就是说，当 TiDB 中 ITEM 有新的变更日志时，新的日志将会被插入到 TIDB_TEST_ITEM 表，然后被 STREAM 捕获。
+
+```
+create or replace stream TEST_ITEM_STREAM on table TIDB_TEST_ITEM append_only=true;
+```
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:235]:
```suggestion
2. 为 `TIDB_TEST_ITEM` 创建一个 STREAM，将 `append_only` 设为 `true`，表示仅接收 `INSERT` 事件。创建的 STREAM 可以实时捕获 `TIDB_TEST_ITEM` 的 `INSERT` 事件，也就是说，当 TiDB 中 `ITEM` 有新的变更日志时，变更日志将会被插入到 `TIDB_TEST_ITEM` 表，然后被 STREAM 捕获。

    ```
    create or replace stream TEST_ITEM_STREAM on table TIDB_TEST_ITEM append_only=true;
    ```
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 6]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+      update set n.i_data = v:i_data, n.i_im_id = v:i_im_id, n.i_name = v:i_name, n.i_price = v:i_price 
+  
+  -- 如果 TEST_ITEM 表中不存在匹配 i_id 的记录，则插入这条记录
+  when not matched then 
+      insert 
+          (i_data, i_id, i_im_id, i_name, i_price) 
+      values 
+          (v:i_data, v:i_id, v:i_im_id, v:i_name, v:i_price)
+;
+```
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:262]:
```suggestion
3. 处理 STREAM 中的数据，根据不同的事件类型，在 `TEST_ITEM` 表中插入、更新或删除 STREAM 数据。

    ```
    --将数据合并到 TEST_ITEM 表
    merge into TEST_ITEM n 
      using 
          -- 查询 TEST_ITEM_STREAM
          (SELECT RECORD_METADATA:key as k, RECORD_CONTENT:val as v from TEST_ITEM_STREAM) stm 
          -- 以 i_id 相等为条件将流和表做匹配
          on k:i_id = n.i_id 
      -- 如果 TEST_ITEM 表中存在匹配 i_id 的记录，并且 v 为空，则删除这条记录
      when matched and IS_NULL_VALUE(v) = true then 
          delete 
      
      -- 如果 TEST_ITEM 表中存在匹配 i_id 的记录，并且 v 不为空，则更新这条记录
      when matched and IS_NULL_VALUE(v) = false then 
          update set n.i_data = v:i_data, n.i_im_id = v:i_im_id, n.i_name = v:i_name, n.i_price = v:i_price 
  
      -- 如果 TEST_ITEM 表中不存在匹配 i_id 的记录，则插入这条记录
      when not matched then 
          insert 
              (i_data, i_id, i_im_id, i_name, i_price) 
          values 
              (v:i_data, v:i_id, v:i_im_id, v:i_name, v:i_price)
    ;
    ```
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 7]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+
+在上面的语句中，我们使用了 Snowflake 的 MERGE INTO 语句，这个语句可以根据条件将流和表做匹配，然后根据不同的匹配结果，执行不同的操作，比如删除、更新或插入。在这个例子中，我们使用了三个 WHEN 子句，分别对应了三种情况：
+
+- 当流和表匹配时，且流中的数据为空，则删除表中的记录
+- 当流和表匹配时，且流中的数据不为空，则更新表中的记录
+- 当流和表不匹配时，则插入流中的数据
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:268]:
```suggestion
    在上面的语句中，我们使用了 Snowflake 的 MERGE INTO 语句，这个语句可以根据条件将流和表做匹配，然后根据不同的匹配结果，执行不同的操作，比如删除、更新或插入。在这个例子中，我们使用了三个 WHEN 子句，分别对应了三种情况：

    - 当流和表匹配时，且流中的数据为空，则删除表中的记录
    - 当流和表匹配时，且流中的数据不为空，则更新表中的记录
    - 当流和表不匹配时，则插入流中的数据
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 8]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+      delete 
+  when matched and IS_NULL_VALUE(v) = false then 
+      update set n.i_data = v:i_data, n.i_im_id = v:i_im_id, n.i_name = v:i_name, n.i_price = v:i_price 
+  when not matched then 
+      insert 
+          (i_data, i_id, i_im_id, i_name, i_price) 
+      values 
+          (v:i_data, v:i_id, v:i_im_id, v:i_name, v:i_price)
+;
+```
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:297]:
```suggestion
4. 周期性执行第三步中的语句，以保证数据的实时性。可通过 Snowflake 的 `SCHEDULED TASK` 来实现：

    ```
    -- 创建一个 TASK，周期性执行 MERGE INTO 语句
    create or replace task STREAM_TO_ITEM
        warehouse = test
        -- 每分钟执行一次
        schedule = '1 minute' 
    when
        -- 当 TEST_ITEM_STREAM 中无数据时跳过
        system$stream_has_data('TEST_ITEM_STREAM') 
    as
    -- 将数据合并到 TEST_ITEM 表，和上文中的 merge into 语句相同
    merge into TEST_ITEM n 
      using 
          (select RECORD_METADATA:key as k, RECORD_CONTENT:val as v from TEST_ITEM_STREAM) stm 
          on k:i_id = n.i_id 
      when matched and IS_NULL_VALUE(v) = true then 
          delete 
      when matched and IS_NULL_VALUE(v) = false then 
          update set n.i_data = v:i_data, n.i_im_id = v:i_im_id, n.i_name = v:i_name, n.i_price = v:i_price 
      when not matched then 
          insert 
              (i_data, i_id, i_im_id, i_name, i_price) 
          values 
              (v:i_data, v:i_id, v:i_im_id, v:i_name, v:i_price)
    ;
    ```
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 9]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
...
+
+至此，我们已经建立了一条具备一定 ETL 能力的数据通路，使得 TiDB 的增量数据变更日志能够被输出到 Snowflake 并且维护一个 TiDB 表的数据副本。这样，我们就可以在 Snowflake 中使用 TiDB 表的数据了。下面只剩下一个小工作，定期清理 TIDB_TEST_ITEM 表中的无用数据：
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:299]:
```suggestion
至此，你就建立了一条具备一定 ETL 能力的数据通路，使得 TiDB 的增量数据变更日志能够被输出到 Snowflake，并且维护一个 TiDB 表的数据副本，实现在 Snowflake 中使用 TiDB 表的数据。最后一步操作是定期清理 `TIDB_TEST_ITEM` 表中的无用数据：
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 10]
```diff
@@ -188,7 +188,126 @@ Snowflake 是一种云原生数据仓库。借助 Confluent 的能力，你只
 
     ![Data preview](/media/integrate/data-preview.png)
 
-6. 在 Snowflake 控制面板中，选择 **Data** > **Database** > **TPCC** > **TiCDC**，可以观察到 TiDB 中的增量数据实时同步到了 Snowflake，如上图。至此，就完成了 TiDB 与 Snowflake 的数据集成。
+6. 在 Snowflake 控制面板中，选择 **Data** > **Database** > **TPCC** > **TiCDC**，可以观察到 TiDB 中的增量数据实时同步到了 Snowflake，如上图。但 Snowflake 中的表结构和 TiDB 中的表结构不同，数据也以“追加”的方式插入 Snowflake 表。在大多数业务场景中，都希望 Snowflake 中的表数据是 TiDB 表的一个副本，而不是存储 TiDB 表的变更日志。下一章节对这个问题进行了解决。
```
Comment (shichun-0415) [inline ticdc/integrate-confluent-using-ticdc.md:191]:
```suggestion
6. 在 Snowflake 控制面板中，选择 **Data** > **Database** > **TPCC** > **TiCDC**，可以观察到 TiDB 中的增量数据实时同步到了 Snowflake，如上图。但 Snowflake 中的表结构和 TiDB 中的表结构不同，数据也以“追加”的方式插入 Snowflake 表。在大多数业务场景中，都希望 Snowflake 中的表数据是 TiDB 表的一个副本，而不是存储 TiDB 表的变更日志。该问题将在下一章节解决。
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**